test/Prompt/DontWrap.C

```diff
@@ -172,7 +172,7 @@ const Test2::Nested::Value::type & Test2::Nested::reference() const {
 Test2::Nested Nest2(80);
 // CHECK: Nested::Nested(160)
 Nest2.simpleAdd(3)
-// CHECK: (Test2::Nested::Value::type) 83
+// CHECK: (int) 83
```

Comment:
Its the way the type is deduced when printing.
template <class T> class vector {
  typedef T value_type;
};
typedef std::vector<int> ivec;

The only difference between the two types (semantically?) is one has a parent in a class, the other not. I've tried to get around it if you check that parent of a typedef is TranslationUnit, Namespace, LinkageSpec: print the typedef type; otherwise print the underlying type.
But then (on libc++ at least) since std::vector<int>::iterator  is std::_1::__wrap_iter<int*> it's parent is a namespace, and prints less good.